ArduPlane/systemid.cpp

```diff
@@ -0,0 +1,294 @@
+            attitude_control->bf_feedforward(false);
+            break;
+        case AxisType::RATE_ROLL:
+            attitude_control->rate_bf_roll_sysid(radians(waveform_sample));
+            break;
+        case AxisType::RATE_PITCH:
+            attitude_control->rate_bf_pitch_sysid(radians(waveform_sample));
+            break;
+        case AxisType::RATE_YAW:
+            attitude_control->rate_bf_yaw_sysid(radians(waveform_sample));
+            break;
+        case AxisType::MIX_ROLL:
+            attitude_control->actuator_roll_sysid(waveform_sample);
+            break;
+        case AxisType::MIX_PITCH:
+            attitude_control->actuator_pitch_sysid(waveform_sample);
+            break;
+        case AxisType::MIX_YAW:
+            attitude_control->actuator_yaw_sysid(waveform_sample);
+            break;
+        case AxisType::MIX_THROTTLE:
+            throttle_offset += waveform_sample;
+            break;
+    }
+
+    // reduce control in XY axis when in position controlled modes
+    plane.quadplane.pos_control->set_xy_control_scale_factor(xy_control_mul);
+
+    log_data();
+}
+
+// @LoggerMessage: SIDD
+// @Description: System ID data
+// @Field: TimeUS: Time since system startup
+// @Field: Time: Time reference for waveform
+// @Field: Targ: Current waveform sample
+// @Field: F: Instantaneous waveform frequency
+// @Field: Gx: Delta angle, X-Axis
+// @Field: Gy: Delta angle, Y-Axis
+// @Field: Gz: Delta angle, Z-Axis
+// @Field: Ax: Delta velocity, X-Axis
+// @Field: Ay: Delta velocity, Y-Axis
+// @Field: Az: Delta velocity, Z-Axis
+
+// log system id
+void AP_SystemID::log_data() const
+{
+#if HAL_LOGGING_ENABLED
+    Vector3f delta_angle;
+    float delta_angle_dt;
+    plane.ins.get_delta_angle(delta_angle, delta_angle_dt);
+
+    Vector3f delta_velocity;
+    float delta_velocity_dt;
+    plane.ins.get_delta_velocity(delta_velocity, delta_velocity_dt);
+
+    if (is_positive(delta_angle_dt) && is_positive(delta_velocity_dt)) {
+        const float dt_ang_inv = 1.0 / delta_angle_dt;
+        const float dt_vel_inv = 1.0 / delta_velocity_dt;
+        AP::logger().WriteStreaming("SIDD", "TimeUS,Time,Targ,F,Gx,Gy,Gz,Ax,Ay,Az",
+                                    "ss-zkkkooo", "F---------", "Qfffffffff",
+                                    AP_HAL::micros64(),
+                                    waveform_time, waveform_sample, waveform_freq_rads / (2 * M_PI),
+                                    degrees(delta_angle.x * dt_ang_inv),
+                                    degrees(delta_angle.y * dt_ang_inv),
+                                    degrees(delta_angle.z * dt_ang_inv),
+                                    delta_velocity.x * dt_vel_inv,
+                                    delta_velocity.y * dt_vel_inv,
+                                    delta_velocity.z * dt_vel_inv);
+
+        // log attitude controller at the same rate
+        plane.Log_Write_Attitude(true);
```

Comment:
change ATT to be lower rate

Reply:
Removed system ID logging of this message.  reverts back to original logging rates